sdv/lite/tabular.py

```diff
@@ -42,22 +47,50 @@ def __init__(self, optimize_for=None, metadata=None):
         if optimize_for == SPEED_PRESET:
             self._model = GaussianCopula(
                 table_metadata=metadata,
-                categorical_transformer='categorical',
+                categorical_transformer='label_encoding',
                 default_distribution='gaussian',
                 rounding=None,
             )
+
+            dtype_transformers = {
+                'i': rdt.transformers.NumericalTransformer(
+                    dtype=np.int64, null_column=False),
+                'f': rdt.transformers.NumericalTransformer(
+                    dtype=np.float64, null_column=False),
+                'O': rdt.transformers.CategoricalTransformer(fuzzy=True),
+                'b': rdt.transformers.BooleanTransformer(null_column=False),
+                'M': rdt.transformers.DatetimeTransformer(null_column=False),
+            }
+            self._model._metadata._dtype_transformers.update(dtype_transformers)
+
             print('This config optimizes the modeling speed above all else.\n\n'
                   'Your exact runtime is dependent on the data. Benchmarks:\n'
                   '100K rows and 100 columns may take around 1 minute.\n'
                   '1M rows and 250 columns may take around 30 minutes.')
 
     def fit(self, data):
         """Fit this model to the data."""
+        self._null_percentages = {}
+        table_meta = self._model._metadata.to_dict()
+
+        for field in table_meta['fields']:
+            num_nulls = data[field].isna().sum()
+            if num_nulls > 0:
+                # Store null percentage for future reference.
+                self._null_percentages[field] = num_nulls / len(data)
+
         self._model.fit(data)
 
     def sample(self, num_rows):
         """Sample rows from this table."""
-        return self._model.sample(num_rows)
+        sampled = self._model.sample(num_rows)
+
+        if self._null_percentages:
+            for field, percentage in self._null_percentages.items():
+                sampled[field] = sampled[field].apply(
+                    lambda x: np.nan if random.random() < percentage else x)
```

Comment:
I think `apply` can sometimes be slow. Can you try making a mask from the columns based on the null percentage for each one and applying the mask to the `DataFrame` instead? I think that should be faster. Here is an example https://cmdlinetips.com/2019/05/how-to-randomly-add-nan-to-pandas-dataframe/, although in this case the percentage of nulls is different for each field

Reply:
Cool, sounds good! I'll update it